src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp

```diff
@@ -820,31 +821,91 @@ HRESULT CorProfiler::ProcessReplacementCalls(
       const auto original_argument = pInstr->m_Arg32;
       const void* module_version_id_ptr = &module_metadata->module_version_id;
 
-      // insert the opcode and signature token as
-      // additional arguments for the wrapper method
-      //
-      // IMPORTANT: Conditional branches may jump to this call instruction, so
-      // we cannot add the argument-loading instructions BEFORE this instruction,
-      // otherwise this will surface in an InvalidProgramException.
-      // Either begin loading the additional arguments starting with this ILInstr
-      // structure or make the current ILInstr a no-op and do all of the argument
-      // loading after this instruction.
+      // Begin IL Modification
       ILRewriterWrapper rewriter_wrapper(&rewriter);
       rewriter_wrapper.SetILPosition(pInstr);
+
+      // IL Modification #1: Replace original method call with a NOP, so that all original
+      //                     jump targets resolve correctly and we correctly populate the
+      //                     stack with additional arguments
+      //
+      // IMPORTANT: Conditional branches may jump to the original call instruction which
+      // resulted in the InvalidProgramException seen in
+      // https://github.com/DataDog/dd-trace-dotnet/pull/542. To avoid this, we'll do
+      // the rest of our IL modifications AFTER this instruction.
       auto original_methodcall_opcode = pInstr->m_opcode;
       pInstr->m_opcode = CEE_NOP;
+      pInstr = pInstr->m_pNext;
+      rewriter_wrapper.SetILPosition(pInstr);
 
-      // replace with a non-virtual call (CALL) to the instrumentation wrapper
-      // always use CALL because the wrappers methods are all static
-      rewriter_wrapper.CallMemberAfter(wrapper_method_ref, false);
```

Comment:
@zacharycmontoya Why is the `CallMemberAfter` removed?

Reply:
This got moved down to a `CallMember` invocation, so it's still there! I just wanted to re-order some of the invocations into a way I thought was more intuitive, and it included removing the only use of `CallMemberAfter` and changing it to a `CallMember`